modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperCppEngine.java

```diff
@@ -0,0 +1,417 @@
+      logger.debug("WhisperC++ number of processors set to {}", whispercppProcessors);
+    }
+
+    whispercppMaxContext = OsgiUtil.getOptCfgAsInt(cc.getProperties(), WHISPERCPP_MAX_CONTEXT_CONFIG_KEY);
+    if (whispercppMaxContext.isSome()) {
+      logger.debug("WhisperC++ max context set to {}", whispercppMaxContext);
+    }
+
+    whispercppSplitOnWord = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_SPLIT_ON_WORD_CONFIG_KEY);
+    if (whispercppSplitOnWord.isSome()) {
+      logger.debug("WhisperC++ split on word set to {}", whispercppSplitOnWord);
+    }
+
+    whispercppBestOf = OsgiUtil.getOptCfgAsInt(cc.getProperties(), WHISPERCPP_BEST_OF_CONFIG_KEY);
+    if (whispercppBestOf.isSome()) {
+      logger.debug("WhisperC++ best of set to {}", whispercppBestOf);
+    }
+
+    whispercppWordThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_WORD_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppWordThreshold.isSome()) {
+      logger.debug("WhisperC++ word threshold set to {}", whispercppWordThreshold);
+    }
+
+    whispercppEntropyThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_ENTROPY_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppEntropyThreshold.isSome()) {
+      logger.debug("WhisperC++ entropy threshold set to {}", whispercppEntropyThreshold);
+    }
+
+    whispercppLogProbThreshold = OsgiUtil.getOptCfg(cc.getProperties(), WHISPERCPP_LOG_PROB_THRESHOLD_CONFIG_KEY).bind(
+        Strings.toDouble);
+    if (whispercppLogProbThreshold.isSome()) {
+      logger.debug("WhisperC++ log prob threshold set to {}", whispercppLogProbThreshold);
+    }
+
+    whispercppDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_DIARIZATION_CONFIG_KEY);
+    if (whispercppDiarization.isSome()) {
+      logger.debug("WhisperC++ diarization set to {}", whispercppDiarization);
+    }
+
+    whispercppTinyDiarization = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_TINY_DIARIZATION_CONFIG_KEY);
+    if (whispercppTinyDiarization.isSome()) {
+      logger.debug("WhisperC++ tiny diarization set to {}", whispercppTinyDiarization);
+    }
+
+    whispercppNoFallback = OsgiUtil.getOptCfgAsBoolean(cc.getProperties(), WHISPERCPP_NO_FALLBACK_CONFIG_KEY);
+    if (whispercppNoFallback.isSome()) {
+      logger.debug("WhisperC++ no fallback set to {}", whispercppNoFallback);
+    }
+
+    logger.debug("Finished activating/updating speech-to-text service");
+  }
+
+  //TODO: Add method for language detection
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.speechtotext.api.SpeechToTextEngine#generateSubtitlesFile(File, File, String, Boolean)
+   */
+
+  @Override
+  public Map<String, Object> generateSubtitlesFile(File mediaFile,
+      File preparedOutputFile, String language, Boolean translate)
+          throws SpeechToTextEngineException {
+
+    if (!FilenameUtils.isExtension(mediaFile.getPath(), "wav")) {
```

Comment:
Not a big deal, but maybe to be a bit more flexible:
```suggestion
    if (!FilenameUtils.isExtension(mediaFile.getPath(), "wav")) {
    if (!mediaFile.getPath().toLowerCase().endsWith(".wav")) {
```